lib/metrics.py

```diff
@@ -124,51 +125,53 @@ def __compare_ssim(planes):
   a, b = planes
   if a is None or b is None: # handle Y800 case
     return 1.0
-  return skimage.measure.compare_ssim(a, b, win_size = 3)
+  return skimage.metrics.structural_similarity(a, b, win_size = 3)
 
 @timefn("ssim")
 def calculate_ssim(filename1, filename2, width, height, nframes = 1, fourcc = "I420", fourcc2 = None):
-  return RawMetricAggregator(min).calculate(
+  return list(map(float, RawMetricAggregator(min).calculate(
     RawFile(filename1, width, height, nframes, fourcc),
     RawFile(filename2, width, height, nframes, fourcc2 or fourcc),
-    nframes, __compare_ssim)
+    nframes, __compare_ssim)))
 
 def __compare_psnr(planes):
   a, b = planes
   if (a == b).all():
     # Avoid "Warning: divide by zero encountered in double_scalars" generated
     # by skimage.measure.compare_psnr when a and b are exactly the same.
     return 100
-  return skimage.measure.compare_psnr(a, b)
+  with expected_warnings(['DEPRECATED']):
+    return skimage.measure.compare_psnr(a, b)
 
 @timefn("psnr")
 def calculate_psnr(filename1, filename2, width, height, nframes = 1, fourcc = "I420"):
-  return RawMetricAggregator(min).calculate(
+  return list(map(float, RawMetricAggregator(min).calculate(
     RawFile(filename1, width, height, nframes, fourcc),
     RawFile(filename2, width, height, nframes, fourcc),
-    nframes, __compare_psnr)
+    nframes, __compare_psnr)))
 
 def __compare_mse(planes):
   a, b = planes
   return skimage.measure.compare_mse(a, b)
 
 @timefn("mse")
 def calculate_mse(filename1, filename2, width, height, nframes = 1, fourcc = "I420"):
-  return RawMetricAggregator(max).calculate(
+  return list(map(float, RawMetricAggregator(max).calculate(
     RawFile(filename1, width, height, nframes, fourcc),
     RawFile(filename2, width, height, nframes, fourcc),
-    nframes, __compare_mse)
+    nframes, __compare_mse)))
 
 def __compare_nrmse(planes):
   a, b = planes
-  return skimage.measure.compare_nrmse(a, b, norm_type = "Euclidean")
+  with expected_warnings(['DEPRECATED']):
+    return skimage.measure.compare_nrmse(a, b, norm_type = "Euclidean")
 
 @timefn("nrmse")
 def calculate_nrmse(filename1, filename2, width, height, nframes = 1, fourcc = "I420"):
-  return RawMetricAggregator(max).calculate(
+  return list(map(float, RawMetricAggregator(max).calculate(
```

Comment:
It would be better to have RawMetricAggregator::calculate function return this result type.

Reply:
Yes, changed RawMetricAggregator::calculate return list

Reply:
Another PR for  this: https://github.com/intel/vaapi-fits/pull/145, please help to review it.